lib/rspec/support/differ.rb

```diff
@@ -178,12 +181,7 @@ def object_to_string(object)
         object = @object_preparer.call(object)
         case object
         when Hash
-          object.keys.sort_by { |k| k.to_s }.map do |key|
-            pp_key   = PP.singleline_pp(key, "")
-            pp_value = PP.singleline_pp(object[key], "")
-
-            "#{pp_key} => #{pp_value},"
-          end.join("\n")
+          ObjectInspector.inspect(object)
```

Comment:
I did a bit of an investigation and here's what I found:

The diff view is only provided on things containing newline characters in them.
Hashes are given newline characters as we map over the key/value pairs and join each of them by them.

`PP.pp` will also add newline characters of it's own accord, if the length exceeds 79 characters.

So an Array **will** appear in the diff view if it:
 - does not contain procs
 - does not contain all numbers
 - will have a total length of more than 79 characters when represented as a string
 - any of it's string elements contain a new line character.

Running this test:
``` ruby
    expected = ["poo", "thot", "that"]
    actual = ["poo\n", "thot", "that", "car"]
    expect(expected).to eq actual
```
Will provide this output in the differ:
``` patch
       @@ -1,5 +1,4 @@
       -poo\n
       +poo
        thot
        that
       -car
``` 

Moving forward, I'm not really sure what we want the behaviour to be.

I also played around with mapping over the elements in an Array and running them through the ObjectInspector, before passing the new array to PP.pp, but it gave me all sorts of issues surrounding double quoted strings and escaping slashes.